dev-dotnet/dotnet-sdk/dotnet-sdk-5.0.203.ebuild

```diff
@@ -0,0 +1,224 @@
+	!system-bootstrap? ( ${BOOT_SRC_URI} )
+"
+S="${WORKDIR}/${MY_P}"
+
+# Using fedora source archive, since Microsoft doesn't provide one.
+# If you wish to build one yourself, see https://github.com/dotnet/source-build/
+# Do note that it isn't currently possible to generate a source archive without
+#  the proprietary "Darc CLI" tool, and downloading a bunch of nuget packages.
+
+# Fedora package script:
+# https://src.fedoraproject.org/rpms/dotnet5.0/blob/rawhide/f/build-dotnet-tarball
+# Or just run ./build-source-tarball.sh from the root of the source-build repo.
+# (the --base-tarball option might be worth using now it's added)
+
+# https://github.com/dotnet/runtime/blob/release/5.0/src/installer/pkg/packaging/deb/dotnet-runtime-deps-debian_config.json#L27
+LICENSE="MIT Apache-2.0 BSD"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="+system-libunwind system-bootstrap"
+RDEPEND="
+	app-crypt/mit-krb5
+	dev-libs/icu
+	dev-util/lttng-ust
+	system-libunwind? ( sys-libs/libunwind )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	dev-util/cmake
+	dev-vcs/git
+	dev-dotnet/dotnet-reference-packages
+	system-bootstrap? ( >=dev-dotnet/dotnet-sdk-${BOOT_PV} )
+"
+
+CHECKREQS_DISK_BUILD="20G"
+CHECKREQS_DISK_USR="1200M"
+
+# Use ninja unless emake is explicitly requested
+# Adapted from cmake.eclass
+case ${CMAKE_MAKEFILE_GENERATOR:-ninja} in
+	emake)
+		BDEPEND+=" sys-devel/make"
+		CMAKE_GENERATOR_FLAG=""
+		;;
+	ninja|*)
+		BDEPEND+=" dev-util/ninja"
+		CMAKE_GENERATOR_FLAG="-ninja"
+		;;
+esac
+
+# If the package can't be built using gcc currently, set this to "true"
+FORCE_CLANG="${FORCE_CLANG:-true}"
+
+if ${FORCE_CLANG}; then
+	BDEPEND+=" sys-devel/clang"
+fi
+
+dotnet_unpack() {
+	if ! use system-bootstrap; then
+		mkdir "${WORKDIR}/dotnet" || die
+		cd "${WORKDIR}/dotnet" || die
+		unpack "dotnet-sdk-${BOOT_PV}-linux-x64.tar.gz"
+		mkdir source-artifacts || die
+		cd source-artifacts || die
+		unpack "Private.SourceBuilt.Artifacts.${BOOT_ARTIFACTS_PV}.tar.gz"
+	fi
+}
+
+dotnet_find() {
+	if ! use system-bootstrap; then
+		DOTNET_ROOT="${WORKDIR}/dotnet"
+		return
+	fi
+
+	for x in "${DOTNET_ROOT}" "${BROOT}/usr/lib/dotnet"; do
+		if [ -d "${x}" ]; then
+			DOTNET_ROOT="${x}"
+		fi
+	done
+	if [ ! -d "${DOTNET_ROOT}" ]; then
+		die "Can't find installed .NET SDK"
+	fi
+}
+
+dotnet_find_reference() {
+	# Handle the case where we're bootstrapping with a binary sdk,
+	#  but the reference-packages have just been installed in /usr/lib
+	for x in "${DOTNET_ROOT}/reference-packages" "${BROOT}/usr/lib/dotnet/reference-packages"; do
+		if [ -d "${x}" ]; then
+			DOTNET_REFERENCE="${x}"
+		fi
+	done
+	if [ ! -d "${DOTNET_REFERENCE}" ]; then
+		die "Can't find installed .NET Source-build Reference Packages"
+	fi
+}
+
+src_unpack() {
+	unpack "${MY_P}.tar.gz"
+	dotnet_unpack
+}
+
+src_prepare() {
+	add_patch() {
+		cp "$2" "patches/$1/9000-$(basename "$2")" || die
```

Comment:
```suggestion
		cp "${2}" "patches/${1}/9000-$(basename "${2}")" || die
```

Reply:
I've been fully going by copying the style and use patterns of other ebuilds/eclasses. It might be useful to make `toolchain-funcs.eclass` comply with this as well, as I was reading that at the time.